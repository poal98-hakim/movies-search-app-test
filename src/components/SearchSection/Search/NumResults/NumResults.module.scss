@use '../../../../scss/abstracts/colors';

.numResultsMovies{
    display: block;
    font-size: .8em;
    color: colors.$PRIMARY_COLOR;
}