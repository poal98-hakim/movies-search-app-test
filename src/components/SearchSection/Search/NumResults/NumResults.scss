@use '../../../../scss/base/colors';

.num-results-movies{
    display: block;
    font-size: .8em;
    color: colors.$PRIMARY_COLOR;
}