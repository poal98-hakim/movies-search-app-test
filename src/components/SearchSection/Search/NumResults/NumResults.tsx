import React from "react";
import "./NumResults.scss";

interface IProps{
    totalResults: number;
}
const NumResults : React.VFC<IProps> = ({totalResults}) => <span className="num-results-movies">{totalResults} results</span>

export default NumResults;
