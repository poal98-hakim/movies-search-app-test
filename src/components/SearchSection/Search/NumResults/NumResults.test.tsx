import NumResults from "./NumResults";
import { render, screen } from '@testing-library/react';

const PROPS = {
  totalResults: 5
}

describe('NumResults', () => {
  let props = {...PROPS};

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<NumResults {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<NumResults {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads total results', () => {
    expect(screen.getByText(`${props.totalResults} results`)).toBeInTheDocument();
  })

})
