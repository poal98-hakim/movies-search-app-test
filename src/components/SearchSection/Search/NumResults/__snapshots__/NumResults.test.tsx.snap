// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumResults renders correctly 1`] = `
<body>
  <div>
    <span
      class="num-results-movies"
    >
      5
       results
    </span>
  </div>
  <div>
    <span
      class="num-results-movies"
    >
      5
       results
    </span>
  </div>
</body>
`;
